airflow/providers/tableau/hooks/tableau.py

```diff
@@ -133,3 +138,28 @@ def get_all(self, resource_name: str) -> Pager:
         except AttributeError:
             raise ValueError(f"Resource name {resource_name} is not found.")
         return Pager(resource.get)
+
+    def get_job_status(self, job_id: str) -> Enum:
+        """
+        Get the current state of a defined Tableau Job.
+        .. see also:: https://tableau.github.io/server-client-python/docs/api-ref#jobs
+
+        :param job_id: The id of the job to check.
+        :type job_id: str
+        :rtype: Enum
+        """
+        return TableauJobFinishCode(int(self.server.jobs.get_by_id(job_id).finish_code))
+
+    def waiting_until_succeeded(self, job_id: str) -> bool:
+        """
+        Wait until the current state of a defined Tableau Job is not PENDING.
+        :param job_id: The id of the job to check.
+        :type job_id: str
+        :return: return True if the job has SUCCESS status, False otherwise.
+        :rtype: bool
+        """
+        finish_code = TableauJobFinishCode.PENDING
+        while finish_code == TableauJobFinishCode.PENDING:
+            finish_code = self.get_job_status(job_id=job_id)
```

Comment:
Do we risk here to overwhelm the Tableau API?

Reply:
Yeah you're right especially for a long refresh. But now I am a little bit confused, the original request was to remove the sensor from the operator and to implement the waiting code in the hook. But why just not use the sensor and the already implemented poke function instead to create a new one?

What do you think?

Reply:
We have the issue of waiting for a specific status in other operators.
For example:
`EC2StartInstanceOperator` allows to pass check_interval to the hook:

https://github.com/apache/airflow/blob/1960e37952efc08ad33328ff511a846f2376ad4f/airflow/providers/amazon/aws/operators/ec2_start_instance.py#L65-L69

The `wait_for_state` in `EC2Hook` has similar functionality to what we want. That way we provide the user the power to decide how much time to wait.

BTW I think it would be nice to change the `waiting_until_succeeded` to a generic`wait_for_state` like it's implemented in the `EC2Hook` that way it will allow more flexibility and users may use it for further custom operators if they need to. For our use case we can wait till`TableauJobFinishCode.SUCCESS`

Reply:
Ok. Now it's more clear. I will implement the changes. 
Thank you.